docs/configuration-ini.md

```diff
@@ -0,0 +1,335 @@
+
+   Type of Network
+     - regression : network for linear regression
+     - knn : K-nearest neighbor
+     - neuralnetwork : Deep Neural Network
+	 
+  
+2. ```learning_rate = <float>```
+
+   Learning rate to decay 
+   
+
+3. ```epoch = <unsigned int>```
+
+   Number of epoch to finish train
+   
+
+4. ```optimizer = <string>```
+
+   Optimizer to update weight and bias
+     - adam : Adaptive Moment Estimation
+     - sgd : stochastic gradient decent
+	 
+  
+5. ```cost = <string>```
+
+   Cost (Loss) function 
+     - mse : mean squared error
+     - cross : cross entropy
+        + sigmoid activation function 
+        + softmax activation function
+		
+	 
+6. ```model = <string>```
+
+   Model file name to save updated weights
+   
+   
+7. ```minibatch = <unsigned int>```
+
+   Mini batch size
+   
+   
+8. ```beta1 = <float>```
+
+   beta1 data for adam optimizer. Only validate for adam.   0.9 is default.
+   
+   
+9. ```beta2 = <float>```
+
+   beta2 data for adam optimizer. Only validate for adam.   0.999 is default.
+   
+   
+10. ```epsilon = <float>```
+
+     Epsilon data for adam optimizer. Only validate for adam.   1.0e-7 is defalut.
+	 
+   
+
+**example**
+
+
+```bash
+# Network Section : Network
+[Network]
+Type = NeuralNetwork
+Learning_rate = 1e-4
+Epoch = 1500
+Optimizer = adam
+Cost = cross
+Model = "model.bin"
+minibatch = 32
+beta1 = 0.9
+beta2 = 0.999
+epsilon = 1e-7
+```
+
+
+### DataSet Section
+Define data set. training, validation, test data set.
+
+Start with "[ DataSet ]"
+
+1. ```buffersize = <unsigned int>```
+
+    Define Buffer size. usually it is grater than mini batch size.
+	
+	
+2. ```traindata = <string>```
+
+    training data file path.   The data must be saved as following format
+	
+	```feature data[i], label data[i], feature data[i+1], label data[i+1], ...```
+	
+	
+3. ```validdata = <string>```
+
+    validation data file path.   The data must be saved as following format
+	
+	```feature data[i], label data[i], feature data[i+1], label data[i+1], ...```
+
+	
+4. ```testdata = <string>```
+
+    test data file path.   The data must be saved as following format
+	
+	```feature data[i], label data[i], feature data[i+1], label data[i+1], ...```
+	
+
+5. ```labeldata = <string>```
+
+    label data file path.   The data must be saved as following format
+	
+	
+
+### Layer Section
+Describe hyper-parameters for layer. Other is matter and it start from top to bottom.
+
+Start with "[ ${layer name} ]"
+
+1. ```type = <string>```
+
+   Type of Layer
+     - input : input layer
+     - conv2d : 2D convolution layer
+     - pooling2d : 2D pooling layer
+     - flatten : flatten layer
+     - fully_connected : fully connected layer
+     - batch_normalization : batch normalization layer
+     - activation : activation layer
+	 
+  
+2. ```kernel_size = <unsigned int>,<unsigned int>```
+
+   Kernel size for convolution layer
+   
+  
+3. ```bias_init_zero = <bool>```
+
+   token to set bias zero
```

Comment:
Token to initialize bias with zeroes. Setting to False would initialize bias randomly.